binding.cc

```diff
@@ -952,28 +995,46 @@ namespace zmq {
     if(zmq_socket_monitor(socket->socket_, addr, ZMQ_EVENT_ALL) != -1) {
       socket->monitor_socket_ = zmq_socket (context->context_, ZMQ_PAIR);
       zmq_connect (socket->monitor_socket_, addr);
-
+      socket->timer_interval_ = timer_interval;
+      socket->num_of_events_ = num_of_events;
       socket->monitor_handle_ = new uv_timer_t;
-
       socket->monitor_handle_->data = socket;
 
       uv_timer_init(uv_default_loop(), socket->monitor_handle_);
-      uv_timer_start(socket->monitor_handle_, reinterpret_cast<uv_timer_cb>(Socket::UV_MonitorCallback), timer_interval, timer_interval);
+      uv_timer_start(socket->monitor_handle_, reinterpret_cast<uv_timer_cb>(Socket::UV_MonitorCallback), timer_interval, 0);
     }
 
     NanReturnUndefined();
   }
 
+  void
+  Socket::Unmonitor() {
+    // Make sure we are monitoring
+    if (this->monitor_socket_ == NULL) {
+      return;
+    }
+
+    // Passing NULL as addr will tell zmq to stop monitor
+    zmq_socket_monitor(this->socket_, NULL, ZMQ_EVENT_ALL);
+
+    // Close the monitor socket and stop timer
+    if (zmq_close(this->monitor_socket_) < 0)
+      throw std::runtime_error(ErrorMessage());
```

Comment:
Guys, see line: https://github.com/JustinTulloss/zeromq.node/pull/424/files#diff-0003efb78f2f135c88ca5431b8353b2cR513
I raise a "special" `monitor_error` event to the user. The reason for not using the `error` event is because the `error` event (imho) should be used only for when there are errors related to the "main" socket, and not to the monitoring socket, in my view, error event means that the socket is no more usable and the user has to close it, which is not the case of monitoring error, there we just close the **monitoring** socket, and the user can create a new one by calling to `monitor`.